pkg/hostagent/dns/dns.go

```diff
@@ -319,7 +340,13 @@ func Start(opts ServerOptions) (*Server, error) {
 		}()
```

Comment:
The code for tcp and udp is nearly identical; the only differences being the `TruncateReply` setting and the `dns.Server.Net` field. So maybe this code should be turned into a function:

```
listenAndServe("udp", opts)
listenAndServe("tcp", opts)
```